Display a notice when there are plugins installed or updates available

janfaracik opened the pull request:
Followup to #8186. Adds a notice to the Installed plugins page(s) as well as to the updates page to tell the user when there are no plugins installed/updates available, this is rather than displaying the table. Also disables the search bar.

<img width="1089" alt="image" src="https://github.com/jenkinsci/jenkins/assets/43062514/bdaee44a-266a-41ab-9bd1-33629dbc1f11">

<img width="1087" alt="image" src="https://github.com/jenkinsci/jenkins/assets/43062514/e7c43df4-781c-4680-9dcd-8a57939e237e">

<img width="1077" alt="image" src="https://github.com/jenkinsci/jenkins/assets/43062514/8b22b9ac-74eb-4d21-8052-d2c256b8614d">

### Testing done

* Notice shows when no plugins are installed
* Notice hides when plugins are installed
* Notice shows when no updates are available
* Notice hides when updates are available

### Proposed changelog entries

- Display a notice when plugin updates are available or when there are no plugins installed.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@jenkinsci/sig-ux 

<!-- Comment:
If you need an accelerated review process by the community (e.g., for critical bugs), mention @jenkinsci/core-pr-reviewers.
-->

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```


<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8208"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

janfaracik committed:
Init

janfaracik committed:
Merge branch 'no-updates-notice' into empty-logs

janfaracik committed:
Update index.jelly

janfaracik committed:
Update index.jelly

janfaracik committed:
Remove files

janfaracik committed:
Create component

janfaracik committed:
Add docs for the attributes

janfaracik committed:
Fix

janfaracik committed:
Update search-bar.scss

janfaracik committed:
Merge branch 'master' into no-updates-notice

NotMyFault commented on core/src/main/resources/hudson/PluginManager/updates.jelly:
```diff
@@ -55,183 +56,178 @@ THE SOFTWARE.
+            </table>
+          </j:when>
+          <j:otherwise>
+            <l:notice icon="symbol-up-to-date" text="${%No updates}" />
```
```suggestion
            <l:notice icon="symbol-up-to-date" text="${%No updates available}" />
```
Given the updated presence, this sounds more fitting to me, thoughts?

janfaracik replied:
I like it!

NotMyFault reviewed (commented):
Tests will need a few fixes too, as it seems

janfaracik committed:
Update core/src/main/resources/hudson/PluginManager/updates.jelly

Co-authored-by: Alexander Brandes <brandes.alexander@web.de>

janfaracik committed:
Fix search bar not being enabled

yaroslavafenkin reviewed (commented):
Looks fine security wise

janfaracik committed:
Add notice to Installed plugins pages, remove :has() as its not supported yet

janfaracik committed:
Merge branch 'master' into no-updates-notice

janfaracik commented:
I've updated this to include the Installed plugins page(s) which had slipped my mind before, and I've also removed the usage of `:has()` as it was breaking the tests ([for the time being](https://github.com/HtmlUnit/htmlunit-cssparser/issues/20)). Rereviews and another security look over would be appreciated. Thanks

janfaracik committed:
Fix content

janfaracik committed:
Merge branch 'master' into no-updates-notice

janfaracik committed:
Merge branch 'master' into no-updates-notice

NotMyFault committed:
Merge branch 'master' into no-updates-notice

daniel-beck commented:
For UI change PRs in general, it would be useful to have both "before" and "after" screenshots (not just "after"). For example, <details><summary>this</summary>

<img width="1412" alt="Screenshot 2023-07-27 at 10 46 12" src="https://github.com/jenkinsci/jenkins/assets/1831569/c0ee3384-5099-4022-b7e8-067a7546972a">

</details> is what the plugin manager currently looks like. While the suggested change does look better, the title and description of this PR are misleading. In all three cases, a message is already shown, there's not just a table header.

timja committed:
Merge branch 'master' into no-updates-notice

timja committed:
Merge branch 'master' into no-updates-notice

timja reviewed (approved):
Looks really nice

NotMyFault reviewed (approved):
This looks nice!

timja commented:
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!
